perceval/components/source.py

```diff
@@ -21,66 +21,123 @@
-
-    :param brightness: the brightness of the source defined as the percentage of unique photon generation
-    :param purity: the ratio of time when photon is emitted alone
-    :param purity_model: `"random"` if additional photons are distinguishable, `"indistinguishable"` otherwise
-    :param indistinguishability: indistinguishability parameter as defined by `indistinguishability_model`
-    :param indistinguishability_model: `"homv"` defines indistinguishability as HOM visibility, `"linear"` defines
-        indistinguishability as ratio of indistinguishable photons
-    :param context: gives a local context for source specific features, like `"discernability_tag"`
-    """
     def __init__(self, brightness: float = 1,
-                 purity: float = 1,
-                 purity_model: str = "random",  # Literal["random", "indistinguishable"]
+                 occupation_factor: float = 1,
+                 multiphoton_component: float = None,
+                 multiphoton_model: Literal["distinguishable", "indistinguishable"] = "distinguishable",
+                 purity: float = None,
                  indistinguishability: float = 1,
-                 indistinguishability_model: str = "homv",  # Literal["homv", "linear"]
+                 indistinguishability_model: Literal["homv", "linear"] = "homv",
+                 overall_transmission: float = 1,
                  context: Dict = None) -> None:
+        r"""Definition of a source
+        :param brightness: Number of photons collected per excitation pulse into the first lens.
+        :param occupation_factor: Occupation factor of the QD state.
+        :param multiphoton_component: Second order intensity autocorrelation at zero time delay :math:'g^{(2)}(0)'
+        :param multiphoton_model: 'distinguishable' if additional photons are distinguishable, 'indistinguishable' otherwise
+        :param purity: preserved for back-compatibility if multiphoton_model is not set. :math:'purity = 1-g^{(2)}(0)'
+        :param indistinguishability: indistinguishability parameter as defined by 'indistinguishability_model'
+        :param indistinguishability_model: 'homv' defines indistinguishability as 2-photon mean wavepacket overlap,
+            'linear' defines indistinguishability as ratio of indistinguishable photons
+        :param overall_transmission: Total transmission of the optical system. Can take into account the brightness of the source.
+        :param context: gives a local context for source specific features, like 'discernability_tag'
+        """
+        if multiphoton_component is None:
+            if purity is None:
+                multiphoton_component = 0
+            else:
+                multiphoton_component = 1 - purity
+        else:
+            assert purity is None, "cannot set both purity and multiphoton_component"
+            
+        if occupation_factor is None:
+            if brightness is None:
+                ocupation_factor = 1
+        else:
```

Comment:
Warnin: You have an issue in the indentation. Here the else if for l.60

Reply:
There is also a missing c in occupation factor

Reply:
This lines were unnecessary. I found a fix using "eta = self.overall_transmission*self.brightness/self.occupation_factor". I removed them and set occupation_factor=1 and brightness=1 by default.